iotdb-core/datanode/src/main/java/org/apache/iotdb/db/pipe/connector/protocol/thrift/async/IoTDBThriftAsyncConnector.java

```diff
@@ -157,6 +157,21 @@ public void transfer(TabletInsertionEvent tabletInsertionEvent) throws Exception
       return;
     }
 
+    if (((EnrichedEvent) tabletInsertionEvent).shouldParsePattern()) {
+      for (final TabletInsertionEvent event :
+          tabletInsertionEvent.processRowByRow(
```

Comment:
There can be a more efficient way... Please refer to PipeRawTabletInsertionEvent#convertToTablet

Reply:
changed